### PR TITLE
Client Fixes - Non Breaking Change

### PR DESCRIPTION
**Pull request by metalurgical**
Non-breaking version of https://github.com/torusresearch/tss-client/pull/45.

Update packages to resolve dependency vulnerabilities.
Fix infinite wait cases.
Wait for server responses and check errors when calling /precompute route instead of rushing the servers.
Add handler for precompute_failed socket message to allow for early short circuit on failure.
Add missing consumed assignment.
Remove unresolvable "test" from package.json.
Add null check on socket.
Update readme.
Add local server tests.
Add comments that require further attention in subsequent PRs.

### CODE REVIEW

File: packages/tss-client/src/client.ts [hunk 1]
```diff
@@ -185,78 +195,49 @@ export class Client {
...
+        });
+        // Add listener for completion
+        socket.on("precompute_complete", async (data, cb) => {
+          const { session, party } = data;
+          if (session !== this.session) {
+            this.log(`ignoring message for a different session... client session: ${this.session}, message session: ${session}`);
+            return;
+          }
+          this._precomputeComplete.push(party);
+          if (cb) cb();
```
Comment (himanshuchawla009) [inline packages/tss-client/src/client.ts:226]: any reason why cb is called after updating _precomputeComplete array in success case but its opposite in failure?

File: packages/tss-client/src/client.ts [hunk 2]
```diff
@@ -384,55 +373,55 @@ export class Client {
...
-              recipient: party,
-              msg,
-              hash_only,
-              original_message,
-              hash_algo,
-              ...additionalParams,
-            }),
+      if (party === this.index) {
+        // create signature fragment for this client
+        sigFragments.push(await tss.local_sign(msg, hash_only, this.precomputed_value));
```
Comment (himanshuchawla009) [inline packages/tss-client/src/client.ts:383]: nice refactor, now code is looks cleaner and readable.

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

File: packages/tss-client/src/client.ts (hunk 1 above)
Comment (metalurgical) [reply packages/tss-client/src/client.ts:226]: Thanks, they should be identical, have fixed it.

File: packages/tss-client/src/client.ts (hunk 2 above)
Comment (metalurgical) [reply packages/tss-client/src/client.ts:383]: Appreciated :)

**Comment by metalurgical** [comment]
close #45

**Review by himanshuchawla009 — APPROVED** [review]
lgtm

**Pull request merged**